### DISCUSSION TITLE
Does having more edges on a GNN helps learning?

### DISCUSSION THREAD

**Discussion by ramsey-coding** [pyg-team/pytorch_geometric#5150] [Q&A]
I am doing a machine translation task using a Graph2Seq graph neutral network.

I am using GCN as my encoder.

Graph stats:

- I have around 400 nodes in the graph per data point.
- In the current graph, on average, a node is connected to 3.5 neighbours.

Potentially, I could add semantic edges between nodes.

But logically does adding more edges help a GNN model?

**Comment by rusty1s** [comment]
This is impossible to tell TBH, and depends on whether these edges might be important to solve the underlying task. Might be good to simply try out. If the added edges just introduce additional noise, then model performance might decrease. In addition, adding more edges may potentially result in over-smoothing.

If you are facing different types of edges you want to add, then, in theory, a GNN model can learn to ignore certain edge types. However, this does not necessarily hold for adding more edges of the same type.
